src/Modules/OrchardCore.Commerce/Services/ShoppingCartHelpers.cs

```diff
@@ -94,6 +95,14 @@ public class ShoppingCartHelpers : IShoppingCartHelpers
             totals = lines.CalculateTotals().ToList();
         }
 
+        for (var i = 0; i < totals.Count; i++)
+        {
+            var total = totals[i];
+
+            // The value is rounded to avoid storing more precision than what the currency supports.
+            totals[i] = new Amount(Math.Round(total.Value, total.Currency.DecimalPlaces), total.Currency);
+        }
```

Comment:
Just like `lines.CalculateTotals()`, move this code into an extension method so it can be called like `totals.RoundTotals()` or something like that.

Reply:
I'd rather you add an `Amount GetRounded()` method to Amount. (I prefer that name because `Round()` could imply mutating the original value.) We want rounding available for individual amounts too. You can add an extension method for `IList<Amount>` if you want.